source/core/NativeMemory.cs

```diff
@@ -1022,12 +1059,37 @@ public static bool IsModelABlimp(int modelHash)
 			IntPtr modelInfo = FindCModelInfo(modelHash);
 			return GetVehicleStructClass(modelInfo) == VehicleStructClassType.Blimp;
 		}
+		public static bool IsModelASubmarineCar(int modelHash)
+		{
+			IntPtr modelInfo = FindCModelInfo(modelHash);
+			return GetVehicleStructClass(modelInfo) == VehicleStructClassType.SubmarineCar;
+		}
 		public static bool IsModelATrailer(int modelHash)
 		{
 			IntPtr modelInfo = FindCModelInfo(modelHash);
 			return GetVehicleStructClass(modelInfo) == VehicleStructClassType.Trailer;
 		}
 
+		public static bool HasVehicleFlag(int modelHash, VehicleFlag1 flag) => HasVehicleFlagInternal(modelHash, (uint)flag, 0x0);
+		public static bool HasVehicleFlag(int modelHash, VehicleFlag2 flag) => HasVehicleFlagInternal(modelHash, (uint)flag, 0x4);
+		public static bool HasVehicleFlag(int modelHash, VehicleFlag3 flag) => HasVehicleFlagInternal(modelHash, (uint)flag, 0x8);
+		public static bool HasVehicleFlag(int modelHash, VehicleFlag4 flag) => HasVehicleFlagInternal(modelHash, (uint)flag, 0xC);
```

Comment:
It seems weird to me to split these up into 4 separate enums. Shouldn't the following work?:
```c#
public enum VehicleFlag : ulong
{
	Big = 0x2,
	IsVan = 0x20,
	...
	EmergencyService = 0x1 << 0x4,
	AllowsRappel = 0x80 << 0x4,
	...
	IsTank = 0x200 << 0x8,
	HasBulletProofGlass = 0x1000 << 0x8,
	...
	HasLowriderHydraulics = (ulong)0x800000 << 0xC,
	HasLowriderDonkHydraulics = (ulong)0x8000000 << 0xC,
}
```

Reply:
You can't assemble 4 `uint` enums into 1 `ulong` one because the number of bytes a `ulong` variable occupies is just twice as that of a `uint`. Maybe there are another solutions, but not possible with the enum type if I missed something important.

Reply:
Unfortunately, you can't properly define enum values using bit shift operators in C#. I didn't think they actually don't work properly when defining enum values, though.

Reply:
I'm well aware that 4 uints don't fit into an ulong, but it looked like it would all fit into 64-bits. I misread the offset calculation though, which is in bytes obviously, not bits, so the bit shifts were not equivalent. So please ignore what I said before.